scripts/runner.node.mjs

```diff
@@ -810,6 +811,12 @@ function isJavaScript(path) {
  * @returns {boolean}
  */
 function isTest(path) {
+  if (path.includes("/test-cluster-") && path.endsWith(".js")) return true;
```

Comment:
It would be nice to avoid this, because that means running `bun test` vs. this script will cause different list of tests to run.

What if we just rename the upstream files to have the `.test` extension?

Reply:
And would it make sense to change the `/` to `path.sep` or join it with `path.join` to make the test pass on Windows as well?

Reply:
> It would be nice to avoid this, because that means running `bun test` vs. this script will cause different list of tests to run.
> What if we just rename the upstream files to have the `.test` extension?

the files do not work correctly unless run directly with `bun <file>`. getting the files to run in CI is much more important than a global `bun test` since during development most files are run individually anyway. the main issue with this approach is that RunCommand doesnt respond to `--timeout`. does `Bun.spawn` accept an abort signal or is there a CI env var we can use to communicate the timeout?

> And would it make sense to change the `/` to `path.sep` or join it with `path.join` to make the test pass on Windows as well?

yes, and i have this change locally

Reply:
> does `Bun.spawn` accept an abort signal or is there a CI env var we can use to communicate the timeout?

oh the runner is using `node:child_process.spawn()` which does so i can look into this

Reply:
excellent, we do

Reply:
I see there's also a `timeout` option which we already set at 3 minutes, which feels a bit high, even outside this

Reply:
hm timeout doesnt throw an error or set the exit code so i might change this to use AbortController anyway

Reply:
hm i see now runner.node.mjs already has a bunch of extra logic to smooth over this so i wont change it for now